### ISSUE TITLE
Ideas for search commands

### ISSUE THREAD

**Issue by Cloud7050** [AY2324S1-CS2103-W14-3/tp#5]
Currently, AB3's `list` lists all contacts. `find` searches by name, operating on the keywords with `OR` logic.

We could combine the two commands for simplicity without losing any functionality. If really needed, we could look into **aliasing**. Power users may appreciate shortened aliases such as `l`.

TLDR add command aliases.

---

We could also introduce a new search command that operates on keywords with `AND` logic, maybe **`filter`**.

This is not in our V1.2 plan, but these ideas are coming to mind because of week 6's tutorial 2, where I'm hoping to make a PR for a new command to kill two birds with one stone. (Although, `filter` would likely not involve new UI, yet I want to touch on that aspect in the tutorial as well, so maybe I'll work on something like a `role` or `tags` command instead.)

> you can add a different command than the one given in the tutorial.

TLDR add `filter` command.

---

All in all, I was thinking we could make such searching more generalised, instead of only being able to operate on names. For example, we could support `find`/`filter` by tag, which would become useful for listing people of a certain category.

In the context of _ConText_, eg a use case could be a user tagging all their profs/tutors/friends/classmates etc for a module with the same module tag, and then searching by that tag to see them all easily.

TLDR support the various flags in search commands.

**Comment by Cloud7050** [comment]
I will keep aliasing and the relationship between the current `find` and `list` in the back of my mind. We can depend on user stories for any new commands we add as features. Closing this for now.
